### ISSUE TITLE
Novel view synthesis finetuning setup

### ISSUE THREAD

**Issue by marcinkusz** [facebookresearch/vggt#230]
Hi, awesome paper and codebase.

I am currently trying to reproduce your novel view synthesis finetuning setup from the paper and apply it on the RE10k dataset.

We have very closely mimicked the training procedure of scene-level LVSM without changing the hyperparameters much but using an extended, trained VGGT model. We extended it in the same way you described in the paper using the conv patch embedder to encode Plücker rays, a dpt head to regress the RGB values, and a l2 pixel + perceptual loss with 1.0 and 0.5 loss weights respectively. We leave all of the other VGGT losses out of the training.

We are able to see consistent improvements and relatively decent results, which is awesome, but we are not sure if our setup matches yours identically. Specifically, we are worried that the finetuning is taking quite a while. The loss is going down, but quite slowly. Moreover, it immediately starts at an extremely low value (smoothened L2 goes from 0.1 to 0.01 in 700 update steps and then starts to heavily stagnate, indicating a LR that's too high?). Below, I have provided a list of parameters and questions that we're not sure about. It would be of extreme help if you could identify how our finetuning setup differs from yours.

Awesome work!

#### Questions:
1. We are currently optimizing for 100k gradient update steps, which seems excessive for finetuning. How long did you train for?
2. We keep the learning rate at $10^{-4}$, which also seems high. What scheduling and approximate LR did you use?
3. Did you freeze the whole model except for the heads or did you keep the alternating attention unfrozen? Did you keep the default VGGT losses (camera, depth ect) active or did you turn them off?
4. We set the weight decay for all values to 0 - mainly because we observed that having it turned on for all parameters forces some of the gradients and parameters to zero out and die. Did you encounter something similar? Which parameters do you leave the weight decay on for?
5. We have relatively rigid memory constraints and only one GPU that can only fit one batch of 4 input views and 1 target view Plücker encoding. We stabilize the gradients a bit by increasing gradient accumulation to 16. Should we consider a different set up or does this seem reasonable?

#### Resulting nvs predictions
Here are some target (left) predicted (right) images from the training that is currently in progress.

![Image](https://github.com/user-attachments/assets/5b98c088-0e59-47b0-92b6-fd848ce3a63a)
![Image](https://github.com/user-attachments/assets/1ea271fc-d726-4866-8ada-c1d14432fb52)
![Image](https://github.com/user-attachments/assets/914d9c82-5fa9-4550-849e-3a9732ea3c3f)

**Comment by jytime** [comment]
1. 100k is too much. I think it would need at most 50k.
2. The lr is determined by effective batch size, i.e., your number of GPUs for training. Our used lr (64 GPUs) may not fit your case.
3. We also finetune the backbone. We turn off the default VGGT losses.
4. I did not play with weight decay in our case. We just used the default value. 
5. I would suggest to accumulate to 64/128 or others. Higher the better (honestly it is a bit brave to try this with just one GPU)

**Comment by Shexiaox** [comment]
Hello, may I ask what values you ultimately set for the parameters accum_steps and max_img_per_gpu? It appears accum_steps is constrained by batch_size and cannot be set too high.
The DynamicBatchSampler computes batch size as: batch_size = max_img_per_gpu / random_image_num
When random_image_num is large (e.g., 24), this results in a small batch size (e.g., 48/24 = 2)
The chunk_batch_for_accum_steps function then tries to split this batch into accum_steps chunks
When accum_steps=4 and batch_size=2, the chunking logic 2 // 4 = 0 produces empty batches
https://github.com/facebookresearch/vggt/issues/340
